plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -717,6 +726,18 @@ func (p *Plugin) getCSIFromPVC(namespace, claimName string) (string, error) {
 	if pv.Spec.CSI != nil {
 		return pv.Spec.CSI.Driver, nil
 	}
+
+	if p.csiTranslator.IsPVMigratable(pv) {
+		// For in-tree PV, we need to convert ("translate") the PV to CSI before checking the driver name.
+		translatedPV, err := p.csiTranslator.TranslateInTreePVToCSI(klog.FromContext(ctx), pv)
+		if err != nil {
+			return "", err
+		}
+		if translatedPV.Spec.CSI != nil {
```

Comment:
do we also need to check `translatedPV != nil`? (I'm not familiar with the TranslateInTreePVToCSI guarantees)

Reply:
we're checking for `pv != nil` before, so I've added the nil check here to be consistent.